src/test/rbd_mirror/test_Instances.cc

```diff
@@ -35,13 +35,15 @@ class TestInstances : public ::rbd::mirror::TestFixture {
 
     void handle(const InstanceIds& instance_ids, Instance* instance) {
       std::unique_lock<std::mutex> locker(lock);
-      for (auto& instance_id : instance_ids) {
-        ceph_assert(instance->count > 0);
-        --instance->count;
-
-        instance->ids.insert(instance_id);
-        if (instance->count == 0) {
-          instance->ctx.complete(0);
+      if(instance->count > 0) {
```

Comment:
How do you get to this point where you can invoke the callback before recording that you expect the callback?